Tests/Fluent.Tests/Compute/SharedGalleryImageTests.cs

```diff
@@ -0,0 +1,337 @@
+
+                    Assert.NotNull(gallery.UniqueName);
+                    Assert.Equal("jvaImageGallery", gallery.Name);
+                    Assert.Equal("jv image gallery", gallery.Description);
+                    Assert.NotNull(gallery.ProvisioningState);
+                    //
+                    // Update the gallery
+                    //
+                    gallery.Update()
+                        .WithDescription("updated java's image gallery")
+                        .WithTag("jdk", "openjdk")
+                        .Apply();
+
+                    Assert.Equal("updated java's image gallery", gallery.Description);
+                    Assert.NotNull(gallery.Tags);
+                    Assert.Equal(1, gallery.Tags.Count);
+                    //
+                    // List galleries
+                    //
+                    IEnumerable<IGallery> galleries = azure.Galleries.ListByResourceGroup(rgName);
+                    Assert.Single(galleries);
+                    galleries = azure.Galleries.List();
+                    Assert.True(galleries.Count() > 0);
+                    //
+                    azure.Galleries.DeleteByResourceGroup(gallery.ResourceGroupName, gallery.Name);
+                }
+                finally
+                {
+                    try
+                    {
+                        azure.ResourceGroups.DeleteByName(rgName);
+                    }
+                    catch { }
+                }
+            }
+        }
+
+
+        [Fact]
+        public void CanCreateUpdateGetDeleteGalleryImage()
+        {
+            using (var context = FluentMockContext.Start(GetType().FullName))
+            {
+                Region region = Region.USWestCentral;
+                string rgName = TestUtilities.GenerateName("vmexttest");
+                string galleryName = TestUtilities.GenerateName("jsim");
+                string galleryImageName = "JavaImages";
+
+                var azure = TestHelper.CreateRollupClient();
+
+                try
+                {
+                    // Create a gallery
+                    //
+                    IGallery javaGallery = azure.Galleries.Define(galleryName)
+                            .WithRegion(region)
+                            .WithNewResourceGroup(rgName)
+                            .WithDescription("java's image gallery")
+                            .Create();
+                    //
+                    // Create an image in the gallery
+                    //
+                    IGalleryImage galleryImage = azure.GalleryImages.Define(galleryImageName)
+                            .WithExistingGallery(javaGallery)
+                            .WithLocation(region)
+                            .WithIdentifier("JavaSDKTeam", "JDK", "Jdk-9")
+                            .WithGeneralizedWindows()
+                            // Optionals - Start
+                            .WithUnSupportedDiskType(DiskSkuTypes.StandardLRS)
```

Comment:
>WithUnSupportedDiskType [](start = 29, length = 23)

this look weird. there is no words `Un Supported` :) Should be just `Unsupported` i.e. `WithUnsupportedDiskType`

Reply:
good catch :)